### PR TITLE
Third lint pass

### PR DESCRIPTION
**Pull request by philtay**
This PR also includes GolangCI-Lint in Travis support.

### CODE REVIEW

File: utils_test.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 
 // Reset random reader to original value
 func resetRandReader() {
-	RandReader = rand.Reader
+	randReader = rand.Reader
```
Comment (csstaub) [inline utils_test.go]: Let's keep this one public. It was exposed on purpose to make it possible to test against deterministic test vectors in client libraries (which I think is a reasonable request/thing to do).

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        - golint
+        - govet
+        - ineffassign
+        - staticcheck
+        - structcheck
+        - stylecheck
+        - unused
+    - path: _test\.go
+      linters:
+        - scopelint
```
Comment (csstaub) [inline .golangci.yml]: Nice!

**Comment by philtay** [comment]
@csstaub done

**Comment by csstaub** [comment]
Thank you @philtay!

**Pull request merged**